### DISCUSSION TITLE
Trimming the level canvas size to selection and/or trimming the level canvas size to the mimimum image boundaries

### DISCUSSION THREAD

**Discussion by Nukiloco** [opentoonz/opentoonz#6570] [Ideas]
**Question**
Is it possible to trim the canvas size to the current selection on a level? Also is it possible to trim the canvas size to the minimum boundaries to anything drawn in a raster level?

**Comment by RodneyBaker** [comment]
Not specifically a means to trim to the current selection, although there are some near equivalents to that.
We likely need to know more about your specific requirement to suggest a specific approach.
A useful approach to resizing (either scaling up or cropping) the current level is to use the 'Canvas Size' option found by Right Clicking on the level in the xsheet/timeline.  This then allows to adjust the size based on the current canvas relative to the location we specify.  Example:
![image](https://user-images.githubusercontent.com/2576432/108593262-608abd00-7338-11eb-8561-e9571ea35508.png)

![image](https://user-images.githubusercontent.com/2576432/108593308-857f3000-7338-11eb-99a9-cf4e969d4e89.png)

(Note:  I would save the level before doing this so you can get back to the previous size easily and not overwrite it)

![image](https://user-images.githubusercontent.com/2576432/108593596-36d29580-733a-11eb-8ebf-5c9c5866e9a6.png)
Here we have the new cropped area overlaid over the old (and then colored).

I suggest this approach first in order to discuss some other means of 'trimming' an image.
For example, at any time we can have the camera 'trim' or fit to the size of a current level
We can do this via Camera Settings:

![image](https://user-images.githubusercontent.com/2576432/108593646-95980f00-733a-11eb-95dc-e33f11ca1e18.png)

Selecting 'Use Current Level Settings' will then move in to that area.
![image](https://user-images.githubusercontent.com/2576432/108593676-c5471700-733a-11eb-8cb6-c5c047f3c6d7.png)

So in this way we could use any level in the xsheet to drive the size of our camera simply by selecting that level and using it to set the Camera.

There are other means/methods as well but those seem to be the most likely to be of use to you at this time.

It does sound like 'Use Current Level Settings' is what you may be after.

**Comment by Nukiloco** [comment]
Basically I want a feature where I can use the Select Tool and crop the Canvas Size to that selection region. I'm assuming this should affect the level size.
Example (using Krita): https://user-images.githubusercontent.com/16618467/108618115-46042280-73d0-11eb-9b02-52ab898eaaac.mp4
I also want a feature where I can trim the current level/cell size so that the Canvas Size can fit what is drawn on the canvas. I'm assuming this should also affect the level size.
Example (using Krita): https://user-images.githubusercontent.com/16618467/108618153-967b8000-73d0-11eb-9c41-f1ca52007eab.mp4

The reason why I want this as a feature is because sometimes I would draw an object on a level/frame but there is too much canvas space left over so I would like to trim down the canvas space though it doesn't seem easy at all to trim down the Canvas Size on the level.

**Comment by RodneyBaker** [comment]
There are some near equivalents to those features as well as approaches to achieve them via workaround but I agree that a set of features related to selection, trimming, etc. would be useful.

I've tagged this report as Feature request and it would be good to describe the desired feature(s) and workflow to the greatest extent possible in order to see those features realized.  Thanks for the video reference as that helps a lot in narrowing down your specific desire.
